framework/wazuh/core/manager.py

```diff
@@ -213,3 +215,31 @@ def replace_in_comments(original_content, to_be_replaced, replacement):
         good_comment = comment.group(2).replace(to_be_replaced, replacement)
         original_content = original_content.replace(comment.group(2), good_comment)
     return original_content
+
+
+def update_api_conf(new_config, node_type):
+    """Update dict and subdicts without overriding unspecified keys and write it in the API.yaml file.
+
+    Parameters
+    ----------
+    new_config : dict
+        Dictionary with the new configuration.
+    node_type : str
+        Type of node (master, worker)
+    """
+    if new_config:
```

Comment:
I don't completely understand the reasoning behind this if else. When can we get an empty new_config?